Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogBehavior.cs

```diff
@@ -0,0 +1,211 @@
+            }
+        }
+
+        private async Task CreatePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!await _designatedChannelService.AnyDesignatedChannelAsync(campaign.GuildId,
+                DesignatedChannelType.PromotionInterface))
+                return;
+
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+
+            var campaignSummary = await _promotionCampaignRepository.GetCampignSummaryByIdAsync(campaign.Campaign.Id);
+
+            var message = await pollChannel.SendMessageAsync(embed: BuildPollEmbed(campaignSummary, 1, 0).Build());
+
+            await message.AddReactionsAsync(new[] {(IEmote)_approveEmoji, _disapproveEmoji});
+
+            var poll = new CachedPromoDialog
+            {
+                MessageId = message.Id,
+                CampaignId = campaignSummary.Id,
+            };
+
+            SetDialogCache(campaign.Campaign.Id, message.Id, poll);
+            await _promotionDialogRepository.CreateAsync(message.Id, campaign.Campaign.Id);
+        }
+
+        private async Task DeletePromoDialog(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(campaign.Campaign.Id, out CachedPromoDialog poll))
+                return;
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+
+            var message =  await pollChannel.GetMessageAsync(poll.MessageId);
+
+            if (!await _promotionDialogRepository.TryDeleteAsync(message.Id))
+                throw new InvalidOperationException("Dialog to be deleted does not exist");
+
+            await message.DeleteAsync();
+        }
+
+        private EmbedBuilder BuildPollEmbed(PromotionCampaignSummary campaign, int Approve, int Oppose, string error = null)
+        {
+                var boldRole = $"**{MentionUtils.MentionRole(campaign.TargetRole.Id)}**";
+                var user = _discordSocketClient.GetUser(campaign.Subject.Id);
+
+                ValueTask<Color> aviColor;
+
+                if ((user.GetAvatarUrl(size: 16) ?? user.GetDefaultAvatarUrl()) is { } avatarUrl)
+                {
+                    aviColor = _imageService.GetDominantColorAsync(new Uri(avatarUrl));
+                }
+                else
+                {
+                    aviColor = new ValueTask<Color>(Color.Teal);
+                }
+
+                var embed = new EmbedBuilder()
+                    .WithTitle($"#{campaign.Id}")
+                    .WithDescription($"**{campaign.Subject.Nickname}** has been nominated for promotion to {boldRole}")
+                    .WithColor(aviColor.Result)
+                    .WithUserAsAuthor(user)
+                    .AddField("The current vote total stands at: ", $" {Approve} {_approveEmoji} / {Oppose} {_disapproveEmoji}")
+                    .WithTimestamp(_utcNow)
```

Comment:
Out of curiosity, does this mean that the timestamp on the embed will always update to display the timestamp of when the most recent vote occurred?

Reply:
yes, i thought it was kind of a cool way to show how active the poll is at a glance.

Reply:
Works for me. Though I wonder if people might confuse it for the timestamp of when the campaign was started and use that to base their idea of when the campaign should end.

Maybe in a future enhancement we could add a background service that updates this embed with a field like "This campaign can be closed." after 48 hours or something

Reply:
Time to close and also some basic participation info are the things i plan on adding after. I didnt add them here because i want feedback from everyone before piling stuff on.